apps/wfo-ui/pages/start-workflow/[workflowName].tsx

```diff
@@ -5,18 +5,21 @@ import { ParsedUrlQuery } from 'querystring';
 
 interface StartWorkFlowPageQuery extends ParsedUrlQuery {
     workflowName: string;
-    productId: string;
+    productId?: string;
+    subscriptionId?: string;
 }
 
 const StartWorkflowPage = () => {
     const router = useRouter();
-    const { workflowName, productId } = router.query as StartWorkFlowPageQuery;
+    const { workflowName, productId, subscriptionId } =
+        router.query as StartWorkFlowPageQuery;
 
-    if (workflowName && productId) {
+    if (workflowName && (productId || subscriptionId)) {
         return (
             <WFOStartWorkflowPage
                 workflowName={workflowName}
                 productId={productId}
+                subscriptionId={subscriptionId}
```

Comment:
productId or subscriptionId are basically the first payload that will be sent to the backend. In WFOStartWorkflowPage you receive these props and push them into an array to be sent to backend. I think it would be nice to replace the props productId and subscriptionId (only in the WFOStartWorkflowPage component) and replace it with something more generic to just pass the data as is to backend as first step.

Reply:
I'm not sure what you mean?

Reply:
I know what you mean now. Made some changes to the typings